File: app/static/javascript/shared/app/models/shift.js
(function(root) {

    "use strict";

    var Models = root.App.Models,
        Shift = Models.Base.extend({
            endpoint: function() {
                var self = this,
                    locationId = self.getUpstreamModelId("locationId"),
                    roleId = self.getUpstreamModelId("roleId")
                ;

                return "locations/" + locationId + "/roles/" + roleId + "/shifts";
            },
        })
    ;

    root.App.Models.Shift = Shift;
})(this);

File: staffjoy/resources/shift.py
from staffjoy.resource import Resource
from staffjoy.resources.shift_eligible_workers import ShiftEligibleWorker


class Shift(Resource):
    PATH = "organizations/{organization_id}/locations/{location_id}/roles/{role_id}/shifts/{shift_id}"
    ID_NAME = "shift_id"

    def get_eligible_workers(self, **kwargs):
        return ShiftEligibleWorker.get_all(self, **kwargs)
